app/merchant/views.py

```diff
@@ -247,8 +278,47 @@ def view_orders():
     else:
         status_filter = None
 
+    ratings = Ratings.query.filter_by(merchant_id=current_user.id).all()
+    rating_dict = {rating.vendor_id: rating for rating in ratings}
+
     return render_template(
         'merchant/orders.html',
         orders=orders.all(),
-        status_filter=status_filter
+        status_filter=status_filter,
+        ratings=rating_dict
     )
+
+
+@merchant.route('/orders/<int:order_id>', methods=['POST'])
+@login_required
+@merchant_required
+def review_orders(order_id):
+    order = Order.query.get(order_id)
+    star_rating = request.json['rating']
+    comment = request.json['review']
+
+    # TODO: flash on orders page
```

Comment:
Okay I found the error on this. The issue is that you are calling this endpoint via AJAX. So if everything is valid, then you will get the small popup in the bottom right saying everything was successful. If it wasn't all right (i.e. if there is no review/0stars), you call a render function for the regular orders page with a flash...which is returned to the AJAX call as a success. Basically the point is that the html template is actually being pass back to your AJAX call. 

It would be better to remake the vendor function as a form.